From 4f2489bdf36ac4d6a340e2a327798a1da394573a Mon Sep 17 00:00:00 2001
From: Yaakov Selkowitz <yselkowitz@users.sourceforge.net>
Date: Wed, 24 Mar 2010 00:08:23 -0500
Subject: [PATCH 13/25] Cygwin/X: Disable unsupported extensions in configure

Several extensions are not supported by XWin, some of which are enabled
by default in configure.  We forcefully disable these early on so that
configure will succeed without arguments and without the corresponding
proto installed.

Signed-off-by: Yaakov Selkowitz <yselkowitz@users.sourceforge.net>
(cherry picked from commit 95c9947f6244d37592958b70760bd1ac885b625a)
---
 configure.ac |   14 +++++++++-----
 1 files changed, 9 insertions(+), 5 deletions(-)

diff --git xorg-server-1.8.0/configure.ac xorg-server-1.8.0/configure.ac
index 1e65c51..3176790 100644
--- xorg-server-1.8.0/configure.ac
+++ xorg-server-1.8.0/configure.ac
@@ -724,8 +724,16 @@ XORG_CHECK_LINUXDOC
 dnl Handle installing libxf86config
 AM_CONDITIONAL(INSTALL_LIBXF86CONFIG, [test "x$INSTALL_LIBXF86CONFIG" = xyes])
 
-dnl XQuartz DDX Detection... Yes, it's ugly to have it here... but we need to handle this early on
+dnl DDX Detection... Yes, it's ugly to have it here... but we need to
+dnl handle this early on so that we don't require unsupported extensions
 case $host_os in
+	cygwin*)
+		DGA=no
+		DRI2=no
+		XF86VIDMODE=no
+		XSELINUX=no
+		XV=no
+		;;
 	darwin*)
 		DRI2=no
 
@@ -1877,10 +1885,6 @@ if test "x$XWIN" = xyes; then
 
 	AC_DEFINE(DDXOSVERRORF, 1, [Use OsVendorVErrorF])
 	AC_DEFINE(DDXBEFORERESET, 1, [Use ddxBeforeReset ])
-	if test "x$XF86VIDMODE" = xyes; then
-		AC_MSG_NOTICE([Disabling XF86VidMode extension])
-		XF86VIDMODE=no
-	fi
 fi
 AM_CONDITIONAL(XWIN, [test "x$XWIN" = xyes])
 AM_CONDITIONAL(XWIN_MULTIWINDOW, [test "x$XWIN" = xyes])
-- 
1.7.0.4

